firebase-config-interop/src/main/java/com/google/firebase/remoteconfig/interop/rollouts/RolloutAssignment.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.remoteconfig.interop.rollouts;
+
+import androidx.annotation.NonNull;
+import com.google.auto.value.AutoValue;
+import com.google.firebase.encoders.DataEncoder;
+import com.google.firebase.encoders.annotations.Encodable;
+import com.google.firebase.encoders.json.JsonDataEncoderBuilder;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/** Model representing a single rollout assigned to an app instance at a point in time. */
+@Encodable
+@AutoValue
+public abstract class RolloutAssignment {
+  private static final String ROLLOUT_ID = "rolloutId";
+  private static final String VARIANT_ID = "variantId";
+  private static final String PARAMETER_KEY = "parameterKey";
+  private static final String PARAMETER_VALUE = "parameterValue";
+  private static final String TEMPLATE_VERSION = "templateVersion";
+
+  /** An ID representing the rollout. The ID is unique within a Remote Config template. */
+  @NonNull
+  public abstract String getRolloutId();
+
+  /** The variant of the rollout assigned to the app instance. */
+  @NonNull
+  public abstract String getVariantId();
+
+  /** The Remote Config parameter key affected by the rollout. */
+  @NonNull
+  public abstract String getParameterKey();
+
+  /** The value of the affected parameter key. */
+  @NonNull
+  public abstract String getParameterValue();
+
+  /** The Remote Config template version containing this rollout. */
+  public abstract long getTemplateVersion();
+
+  public static final DataEncoder ROLLOUT_ASSIGNMENT_JSON_ENCODER =
+      new JsonDataEncoderBuilder().configureWith(AutoRolloutAssignmentEncoder.CONFIG).build();
+
+  /** Create a {@code RolloutAssignment} from its JSON-encoded representation. */
+  public static RolloutAssignment create(String jsonString) throws JSONException {
```

Comment:
Wonder for `RolloutAssignment`, is it possible to have a parameter value length validation function to restrict the max length pass to crashlytics? https://github.com/firebase/firebase-android-sdk/pull/5233/files#diff-442de867f9aec30da44b1c83269c068ef34ac6b6c8332359bba5f0740d28bc8fR72

Reply:
I think we should keep that validation/restriction in the Crashlytics SDK since other consumer SDKs might not have the same requirement

Reply:
Seems the implementation will become a bit redundant then, on Crashlytics side whenever we get the list of RolloutAssignment, we still need to iterate the list of object and re-create a new object for the ones which exceed the limit. If that's the case I don't see the reason we want to use the same object. Or is it possible to provide an api in `RolloutsState` to give other SDK the ability to return the rolloutsState with validation? for example:

```
getRolloutsStateWithValidatedParameterValueLength(int maxLenght): {}
```

Reply:
Discussed offline: we'll merge the PR as-is and add the validation logic here later if it makes sense.